ee/vellum_ee/workflows/display/tests/workflow_serialization/test_scheduled_trigger_serialization.py

```diff
@@ -32,27 +32,37 @@ def mock_metadata_json(trigger_class_name: str, trigger_id: UUID) -> Iterator[Tu
     with tempfile.TemporaryDirectory() as tmpdir:
         # Create metadata.json with trigger mapping
         metadata_path = os.path.join(tmpdir, "metadata.json")
-        trigger_module_path = f"{__name__}.{trigger_class_name}"
+
+        test_module_parts = __name__.split(".")
+        workflow_root_module = ".".join(test_module_parts[:-1])  # Remove last part
+
+        relative_module_path = "." + test_module_parts[-1]
 
         with open(metadata_path, "w") as f:
             json.dump(
                 {
                     "trigger_path_to_id_mapping": {
-                        trigger_module_path: str(trigger_id),
+                        # Use relative module path (just the module, not the class name)
+                        relative_module_path: str(trigger_id),
                     }
                 },
                 f,
             )
 
-        # Mock the workflow root finder to return our temp directory
-        # Mock os.path.exists to return True for our metadata.json
-        with patch("vellum.workflows.triggers.base._find_workflow_root_with_metadata", return_value=tmpdir):
-            with patch("vellum.workflows.triggers.base.os.path.exists", return_value=True):
-                with patch("vellum.workflows.triggers.base.virtual_open", open):
-                    # Clear the LRU cache to ensure fresh read
-                    _get_trigger_path_to_id_mapping.cache_clear()
+        # Create a custom mock for virtual_open that returns our tmpdir file
+        def mock_virtual_open(path):
+            # Regardless of the path requested, return our test metadata.json
+            return open(metadata_path)
 
-                    yield trigger_module_path, trigger_id
+        # Mock the workflow root finder to return the parent module path
+        with patch(
+            "vellum.workflows.triggers.base._find_workflow_root_with_metadata", return_value=workflow_root_module
```

Comment:
[nit] upon re-review, this feels core to what we want to test and worth not mocking. Honestly the `virutal_open` too